src/components/engagement/engagement.service.ts

```diff
@@ -707,7 +715,11 @@ export class EngagementService {
     id: string,
     session: ISession
   ): Promise<LanguageEngagement | InternshipEngagement> {
-    this.logger.debug('readOne', { id, userId: session.userId });
+    this.logger.info('readOne', { id, userId: session.userId });
+
+    if (!id) {
+      throw new NotFoundException('no id given');
+    }
```

Comment:
Not necessary

Reply:
It should error out, right? Without that catch it finds the first engagement and returns that.

Reply:
If id is empty/falsy/not a short id it should give back a validation error before this method is ever called. Is that not what you are experiencing?